scripts/translate_pptx_inplace.py

```
@@ -972,20 +1028,111 @@ def main():
-        out = batch_translate(client, args.model, batch, glossary)
+        
+        # Time the batch translation with retry handling
+        t0 = time.time()
+        batch_success = False
+        attempt = 0
+        
+        while not batch_success and attempt < 3:
+            try:
+                out = batch_translate(client, args.model, batch, glossary)
+                batch_success = True
+            except (openai.APIError, openai.APITimeoutError, openai.RateLimitError) as e:
```

Ruff (0.12.2)

Undefined name `openai`

(F821)